File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/provision/PrepareImageResultToStackEventConverter.java
package com.sequenceiq.cloudbreak.core.flow2.stack.provision;

import com.sequenceiq.cloudbreak.cloud.event.setup.PrepareImageResult;
import com.sequenceiq.cloudbreak.common.event.Payload;
import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
import com.sequenceiq.flow.core.PayloadConverter;

public class PrepareImageResultToStackEventConverter implements PayloadConverter<StackEvent> {
    @Override
    public boolean canConvert(Class<?> sourceClass) {
        return PrepareImageResult.class.isAssignableFrom(sourceClass);
    }

    @Override
    public StackEvent convert(Object payload) {
        return new StackEvent(((Payload) payload).getResourceId());
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/provision/PrepareImageResultToStackEventConverter.kt
package com.sequenceiq.cloudbreak.core.flow2.stack.provision

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformResult
import com.sequenceiq.cloudbreak.cloud.event.setup.PrepareImageResult
import com.sequenceiq.cloudbreak.core.flow2.PayloadConverter
import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent

class PrepareImageResultToStackEventConverter : PayloadConverter<StackEvent> {
    override fun canConvert(sourceClass: Class<*>): Boolean {
        return PrepareImageResult::class.java!!.isAssignableFrom(sourceClass)
    }

    override fun convert(payload: Any): StackEvent {
        return StackEvent((payload as CloudPlatformResult<CloudPlatformRequest<Any>>).request!!.cloudContext.id)
    }
}
